torchmetrics/functional/classification/dice.py

```diff
@@ -111,3 +114,202 @@ def dice_score(
 
         scores[i - bg_inv] += score_cls
     return reduce(scores, reduction=reduction)
+
+
+def _dice_compute(
+    tp: Tensor,
+    fp: Tensor,
+    fn: Tensor,
+    average: str,
+    mdmc_average: Optional[str],
+    zero_division: int = 0,
+) -> Tensor:
+    """Computes dice from the stat scores: true positives, false positives, false negatives.
+
+    Args:
+        tp: True positives
+        fp: False positives
+        fn: False negatives
+        average: Defines the reduction that is applied
+        mdmc_average: Defines how averaging is done for multi-dimensional multi-class inputs (on top of the
+            ``average`` parameter)
+
+    Example:
+        >>> from torchmetrics.functional.classification.stat_scores import _stat_scores_update
+        >>> from torchmetrics.functional.classification.dice import _dice_compute
+        >>> preds  = torch.tensor([2, 0, 2, 1])
+        >>> target = torch.tensor([1, 1, 2, 0])
+        >>> tp, fp, tn, fn = _stat_scores_update(preds, target, reduce='micro')
+        >>> _dice_compute(tp, fp, fn, average='micro', mdmc_average=None)
+        tensor(0.2500)
+    """
+    numerator = 2 * tp
+    denominator = 2 * tp + fp + fn
+
+    if average == AverageMethod.MACRO and mdmc_average != MDMCAverageMethod.SAMPLEWISE:
+        cond = tp + fp + fn == 0
+        numerator = numerator[~cond]
+        denominator = denominator[~cond]
+
+    if average == AverageMethod.NONE and mdmc_average != MDMCAverageMethod.SAMPLEWISE:
+        # a class is not present if there exists no TPs, no FPs, and no FNs
+        meaningless_indeces = torch.nonzero((tp | fn | fp) == 0).cpu()
+        numerator[meaningless_indeces, ...] = -1
+        denominator[meaningless_indeces, ...] = -1
+
+    return _reduce_stat_scores(
+        numerator=numerator,
+        denominator=denominator,
+        weights=None if average != "weighted" else tp + fn,
+        average=average,
+        mdmc_average=mdmc_average,
+        zero_division=zero_division,
+    )
+
+
+def dice(
```

Comment:
can we clarify the difference to `dice_score`?

Reply:
Hi, I can clarify. `dice_score` implementation supports only 'macro' reduction, which means that metric calculates for each class separately. And, by default, in `dice_score` implementation first class (which usually background class) skipped.

So `dice` includes `dice_score`, if `average='macro'` and `ignore_index=0`. I decided not to remove the old function so that users do not experience errors when updating the library version.

But I can describe all this in the documentation:)

Reply:
I see so lets set `dice_score` as deprecated and use `dice` inside...
@SkafteNicki any suggestion on which name we shall keep - `dice_score` or `dice`?

Reply:
I would prefer `dice` as naming

Reply:
also prefer `dice`

Reply:
@Borda Hi, I renamed all `dice_scores` files to `dice`, should I add 'will be deprecated' to docstring in `dice_score` or something else?